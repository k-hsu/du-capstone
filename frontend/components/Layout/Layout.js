import React from "react";
import Header from "../Header/Header";
import Footer from "../Footer/Footer";
import ToastManager from "../../components/ToastManager/ToastManager";

const Layout = ({ children }) => {
  return (
    <>
      <Header />
      {children}
      <Footer />
      <ToastManager />
    </>
  );
};

export default Layout;
