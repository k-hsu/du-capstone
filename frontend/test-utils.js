import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { renderHook } from "@testing-library/react-hooks";

function setup(jsx) {
  return {
    user: userEvent.setup(),
    ...render(jsx),
  };
}

export * from "@testing-library/react";
export { renderHook, setup };
