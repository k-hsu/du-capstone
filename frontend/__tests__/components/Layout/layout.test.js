import React from "react";
import { render, screen, renderer } from "../../../test-utils";
import Layout from "../../../components/Layout/Layout";

describe("Layout", () => {
  it("should render layout component", () => {
    render(<Layout>children</Layout>);
    expect(screen.getByText("children")).toBeInTheDocument();
  });
  it("should render layout component styles", () => {
    const component = renderer.create(<Layout>children</Layout>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
