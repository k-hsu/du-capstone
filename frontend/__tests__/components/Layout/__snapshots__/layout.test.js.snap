// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render layout component styles 1`] = `
[
  <div
    className="css-1m9yeco"
    direction="row"
  >
    <span
      style={
        {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "inline-block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "width": "initial",
        }
      }
    >
      <span
        style={
          {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2732%27%20height=%2732%27/%3e"
          style={
            {
              "background": "none",
              "border": 0,
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        />
      </span>
      <img
        alt="capstone-logo"
        data-nimg="intrinsic"
        decoding="async"
        onError={[Function]}
        onLoad={[Function]}
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          alt="capstone-logo"
          data-nimg="intrinsic"
          decoding="async"
          loading="lazy"
          src="/DU-Logo-Mark.svg"
          srcSet="/DU-Logo-Mark.svg 1x, /DU-Logo-Mark.svg 2x"
          style={
            {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </span>
    <text
      className="css-saonad"
      fontFamily="IBM Plex Sans"
    >
      Kobi's Capstone Project
    </text>
  </div>,
  "children",
  <div
    className="css-1tvnu3w"
    direction="row"
  >
    <text
      className="css-vofh07"
      color="white"
      fontFamily="IBM Plex Sans"
    >
      ©2022 OmniFederal · All Rights Reserved · Hello World
    </text>
  </div>,
]
`;
